mava/environment_loop.py

```diff
@@ -575,28 +580,61 @@ def should_run_loop(eval_condtion: Tuple) -> bool:
                         "Running eval loop at executor step: "
                         + f"{self._last_evaluator_run_t}"
                     )
+
             return should_run_loop
 
         episode_count, step_count = 0, 0
 
-        # Currently, we only use intervals for eval loops.
-        environment_loop_schedule = (
-            self._executor._evaluator and self._executor._interval
-        )
-        if environment_loop_schedule:
-            eval_condition = check_count_condition(self._executor._interval)
+        # TODO (Ruan): Checking whether we are using Jax or TF here
+        # this should be removed once we deprecate TF.
+        if hasattr(self._executor, "store"):
+            environment_loop_schedule = self._executor._evaluator and (
+                self._executor.store.evaluation_interval is not None
+            )
+
+            if environment_loop_schedule:
+                eval_condition = check_count_condition(
+                    self._executor.store.evaluation_interval
+                )
+                evaluation_duration = self._executor.store.evaluation_duration
+        else:
+            environment_loop_schedule = (
+                self._executor._evaluator and self._executor._interval
+            )
+            if environment_loop_schedule:
+                eval_condition = check_count_condition(self._executor._interval)
 
         while not should_terminate(episode_count, step_count):
```

Comment:
As discussed, we should remove this, perhaps in another PR

Reply:
Yes, I think perhaps a separate PR. Ideally we can do this once TF is removed.

Reply:
Agreed